### PR TITLE
Added `type` parameter to `attrs.field()` function

### LINKED ISSUE #1082: Add `type` back to `attrs.field()` for use in `attrs.make_class()`
I'm using `attrs.make_class()` to dynamically construct a class based on other attrs classes, and it would be nice to be able to specify the `type` of the fields on the generated class using `attrs.field()` instead of `attr.ib()`. In other words, this works:

```python
fields = {
    field.name: attr.ib(type=field.type)
    for field in itertools.chain.from_iterable(
        attrs.fields(t) for t in SOME_ATTRS_CLASSES
    )
    if not field.name.startswith("_")
}

MyClass = attrs.make_class("MyClass", fields)
```

but this does not:

```python
fields = {
    field.name: attrs.field(type=field.type)
    for field in itertools.chain.from_iterable(
        attrs.fields(t) for t in SOME_ATTRS_CLASSES
    )
    if not field.name.startswith("_")
}

MyClass = attrs.make_class("MyClass", fields)
```

### PR DESCRIPTION
**Pull request by chrysle**
# Summary

This resolves #1082 

Could somebody help me with the tests? I'm not sure they're right where they are. 

# Pull Request Check List

- [x] Added **tests** for changed code.
  Our CI fails if coverage is not 100%.
- [ ] New features have been added to our [Hypothesis testing strategy](https://github.com/python-attrs/attrs/blob/main/tests/strategies.py).
- [x] Changes or additions to public APIs are reflected in our type stubs (files ending in ``.pyi``).
    - [ ] ...and used in the stub test file `tests/typing_example.py`.
    - [ ] If they've been added to `attr/__init__.pyi`, they've *also* been re-imported in `attrs/__init__.pyi`.
- [x] Updated **documentation** for changed code.
    - [ ] New functions/classes have to be added to `docs/api.rst` by hand.
    - [ ] Changes to the signature of `@attr.s()` have to be added by hand too.
    - [x] Changed/added classes/methods/functions have appropriate `versionadded`, `versionchanged`, or `deprecated` [directives](http://www.sphinx-doc.org/en/stable/markup/para.html#directive-versionadded).
          The next version is the second number in the current release + 1.
          So if the current version on PyPI is 23.1.0, the next version is gonna be 23.2.0.
- [ ] Documentation in `.rst` files is written using [semantic newlines](https://rhodesmill.org/brandon/2012/one-sentence-per-line/).
- [x] Changes (and possible deprecations) have news fragments in [`changelog.d`](https://github.com/python-attrs/attrs/blob/main/changelog.d).
- [x] Consider granting [push permissions to the PR branch](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork), so maintainers can fix minor issues themselves without pestering you.

### CODE REVIEW

File: docs/examples.md [hunk 1]
```diff
@@ -626,11 +626,13 @@ Sometimes you may want to create a class programmatically.
 >>> from attrs import make_class
 >>> @define
 ... class C1:
-...     x = field()
+...     x = field(type=int)
```
Comment (hynek) [inline docs/examples.md:629]: this broke the test in 632; technically it should pass if you add an `: int` annotation to `y`?

Comment (chrysle) [reply docs/examples.md:629]: I think I fixed it. Is this ok?

File: tests/test_next_gen.py [hunk 2]
```diff
@@ -28,6 +28,14 @@ def test_simple(self):
         """
         C("1", 2)
 
+    def test_field_type(self):
+        """
+        Make class with attrs.field and type parameter.
+        """
+        classFields = {"testint": attrs.field(type=int)}
+        A = attrs.make_class("A", classFields)
+        assert int == attrs.fields(A).testint.type
```
Comment (hynek) [inline tests/test_next_gen.py:37]:
```suggestion
        classFields = {"testint": attrs.field(type=int)}

        A = attrs.make_class("A", classFields)

        assert int == attrs.fields(A).testint.type
```

File: src/attr/_next_gen.py [hunk 3]
```diff
@@ -179,6 +180,7 @@ def field(
     Identical to `attr.ib`, except keyword-only and with some arguments
     removed.
 
+    .. versionadded:: 22.3.0 type parameter
```
Comment (hynek) [inline src/attr/_next_gen.py:183]:
```suggestion
    .. versionadded:: 22.3.0
       The *type* parameter has been re-added; mostly for
       {func}`attrs.make_class`. Please note that type checkers ignore this
       metadata.
```

**Review by hynek — APPROVED** [review]
Thanks, I've just added some warnings!

Sorry it took so long, sadly this is about the worst time of the year for me and FOSS.

**Comment by chrysle** [comment]
No problem, thanks for accepting!

**Pull request merged**